src/core/WorkItemBuiltins.cpp

```diff
@@ -356,8 +356,15 @@ namespace oclgrind
       if ((address & 0x3) != 0) {
         workItem->m_context->logError("Unaligned address on atomic_max");
       }
-      uint32_t old = memory->atomic(AtomicMax, address, UARG(1));
-      result.setUInt(old);
+	  bool is_signed = overload.back() == 'i';
+	  if (is_signed) {
+		  int32_t old = memory->atomic<int32_t>(AtomicMax, address, SARG(1));
+		  result.setSInt(old);
+	  }
+	  else {
+		  uint32_t old = memory->atomic<uint32_t>(AtomicMax, address, UARG(1));
+		  result.setUInt(old);
+	  }
     }
```

Comment:
Two spaces for indentation here as well please.

The main style also uses a newline before opening brackets for `if` statements, although admittedly there is a contradiction with the conditional block immediately preceding this one that crept it somehow.